### ISSUE TITLE
Add tensorboard visualization of GAN output in the training process

### ISSUE THREAD

**Issue by abster95** [gordicaleksa/pytorch-gans#1]
I've found that when working with GANs the images that it outputs are usually a much better sign of it properly training than looking at the loss output.
You're already using tensorboard so it should be easy to add something similar to [this call](https://github.com/Petlja/PSIML/blob/master/workshops/gan/src/training.py) (line 128).

**Comment by gordicaleksa** [comment]
The functionality is already in there (I agree I also mostly used images to get an understanding of what's going on I even wrote that remark down in the README just below the losses curve graph, check it out, README contains lots of useful info):
![image](https://user-images.githubusercontent.com/29271842/92461030-00a8bb80-f1c9-11ea-9437-3e2b451bc9c5.png)

I've already used that functionality in some of my other projects, check it out here for example:
https://github.com/gordicaleksa/pytorch-nst-feedforward/blob/master/training_script.py#L89

The reasons behind this design decision were:
1. It's complicated seeing lots of images in tensorboard (slider constraints although there are some commands like "--samples_per_plugin images=50")
2. Resolution (it's easier to control the resolution when you dump the image into the file system - via photo editors, etc.)
3. Hard to create GIFs as I can't extract imagery from tensorboard files (it's super annoying)

**Comment by abster95** [comment]
I prefer having it all in one dashboard instead of running different functions and searching for the output in the file system. I found this is especially helpful for beginners. If you feel like it's unnecessary feel free to close this issue with won't fix.

**Comment by gordicaleksa** [comment]
Just added the support, thanks for the feedback! It's ok to have 2 places to look at the intermediate imagery as that's super important for GAN debugging.
